core/src/main/java/org/mule/runtime/core/internal/metadata/MuleMetadataManager.java

```diff
@@ -100,6 +108,17 @@ public void initialise() throws InitialisationException {
     return exceptionHandledMetadataFetch(componentId, MetadataAware::getMetadataKeys, EXCEPTION_RESOLVING_METADATA_KEYS);
   }
 
+  @Override
+  public MetadataResult<Map<String, Set<MetadataKey>>> getMetadataKeys(ConfigurationId configurationId) {
```

Comment:
Can't we use a ComponentId? I know it has `getFlowName` in its api, but at some point we could migrate to using only the `getComponentPath`. Once we implement #MULE-9496 we can come up with a convention over how a path for a global element looks like vs something inside a flow.

If we don't change this now, at least we should add a `TODO MULE-9496`